Mathlib/MeasureTheory/Constructions/Convolution.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.MeasureTheory.Constructions.Prod.Basic
+import Mathlib.MeasureTheory.Measure.MeasureSpace
+/-!
+# The convolution of measures
+
+In this file we define and prove properties about the convolution of two measures
+
+## Main definition
+
+* `MeasureTheory.Measure.conv`: The convolution of two measures.
+
+## Main results
+
+* add
+## Implementation Notes
+
+## Tags
+
+-/
+
+noncomputable section
+
+namespace MeasureTheory
+
+namespace Measure
+
+/-- Convolutions of measures. They are defined for arbitrary measures on a monoid M that is also
+a measurable space. TODO: should get a to_additive version for AddMonoids -/
+def conv {M : Type*} [Monoid M] [MeasurableSpace M] (μ : Measure M) (ν : Measure M) :
+    Measure M := Measure.map (fun x : M × M ↦ x.1 * x.2) (Measure.prod μ ν)
+
+/-- Scoped notation for the convolution of measures. When a dirac measure is involved,
+this seems to lead to ambiguity for some reason. -/
+scoped infix:80 " * " => conv
```

Comment:
I'm thinking you should do something like [this](https://github.com/leanprover-community/mathlib4/blob/38c547ff15964e9e2251bd7da687ab8465dfb307/Mathlib/MeasureTheory/Measure/MutuallySingular.lean#L45) so that you only get this notation when you open MeasureTheory

Reply:
fixed!